### DISCUSSION TITLE
Linux: compositing Japanese with ibus (mozc) failed with warning

### DISCUSSION THREAD

**Discussion by tomocrafter** [ghostty-org/ghostty#3293] [Issue Triage]
Typing Japanese on Ubuntu with ibus (mozc) failed with following error:

`(ghostty:44988): IBUS-WARNING **: 13:52:46.409: ibus_input_context_post_process_key_event: Type 'h' is not supported.`

Pleaes let me know if I can collect more detailed logs to provide. Thank you!

**Comment by mitchellh** [comment]
We have Japanese users using mozc from the beta. Converted to a discussion to gather more info. Once we have more details we can convert back.

**Comment by mitchellh** [comment]
#3628

**Reply by tomocrafter** [reply]
Technically, I can enable IME but can not confirm composition with Enter key, and I'm on X11 instead of wayland.

Also, I can even type if I use Tab key (or clicking other application to remove focus from ghostty) instead of Enter key to confirm composition.
![image](https://github.com/user-attachments/assets/685d4358-ebd2-4d12-9812-2ed119b42bdb)


Is this issue applies for me as well?

**Reply by tomocrafter** [reply]
@mitchellh 
I tried 1.1.0 release but seems like this issue is not fixed yet.
So I believe it is not same issue as #3628.

**Reply by tomocrafter** [reply]
This is reproduce video with keystrokes. The composition remains on terminal.
In this video, I forgot to mention that same problem happens on wezterm if I removed all characters with backspace while compositing. which I reported to them with https://github.com/wez/wezterm/issues/6433. but compostion itself works on wezterm.
[Screencast from 2025-02-03 10-50-01.webm](https://github.com/user-attachments/assets/b3c71160-476b-4e75-8035-813c5a500990)

**Reply by pluiedev** [reply]
Couldn't reproduce on Wayland + Fcitx 5 + Mozc. So it's probably some weird edge case with either ibus or X11. What is your ibus version? (1.5.29, 1.5.30 and 1.5.31 all exhibit slightly different behavior...)

**Reply by tomocrafter** [reply]
Thank you for investigation.
I'm on X11 + iBus + Mozc with:
```
❯ ibus version
IBus 1.5.29-rc2
```

which is default version for `Ubuntu 24.04.1 LTS`:

```
❯ apt-cache policy ibus
ibus:
  Installed: 1.5.29-2
  Candidate: 1.5.29-2
  Version table:
 *** 1.5.29-2 500
        500 http://jp.archive.ubuntu.com/ubuntu noble/main amd64 Packages
        100 /var/lib/dpkg/status
```

**Reply by tomocrafter** [reply]
I tried ubuntu on wayland and this bug did not occur. but wayland on nvidia is extreamly unstable so I can't use it daily.
Seems like it is not bug of ghostty?

**Reply by pluiedev** [reply]
Possibly. Input methods on Linux are... a hot mess to say the least, and every platform and every input method framework reports slightly different behavior to Ghostty, and the best we can do is to work around it...

**Reply by mitchellh** [reply]
I have a NixOS config that can reproduce this environment but it has to rebuild the entire Gnome ecosystem from scratch so I'll plan to do it... another time.
